quinn/examples/server.rs

```diff
@@ -247,7 +247,7 @@ fn process_get(root: &Path, x: &[u8]) -> Result<Vec<u8>> {
         bail!("missing \\r\\n");
     }
     let x = &x[4..x.len() - 2];
-    let end = x.iter().position(|&c| c == b' ').unwrap_or_else(|| x.len());
+    let end = x.iter().position(|&c| c == b' ').unwrap_or(x.len());
```

Comment:
I think clippy won't like this, since it makes a function call unconditional.

Reply:
The following is free of clippy errors:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=cae14bfe382e66bfdda86fc25b5ff652

More details here:
https://github.com/rust-lang/rust-clippy/issues/6943